### PR TITLE
Add a new literal data type for user ID default

### PR DESCRIPTION
**Pull request by mjgiarlo**
Connects to #3357

## Why was this change made?

...

## How was this change tested?

...

## Which documentation and/or configurations were updated?

...

### CODE REVIEW

File: static/templates/rt_literal_property_attrs_doc.json [hunk 1]
```diff
@@ -100,6 +100,50 @@
...
+      }
+    ],
+    "http://sinopia.io/vocabulary/hasRemark": [
+      {
+        "@value": "Data type specifying a user ID default literal."
+      }
+    ],
+    "http://sinopia.io/vocabulary/hasPropertyUri": [
+      {
+        "@id": "http://sinopia.io/vocabulary/userIdDefaultDataType"
```
Comment (mjgiarlo) [inline static/templates/rt_literal_property_attrs_doc.json:118]: @justinlittman Is this an acceptable value? I made this up. Not sure if this should be the same or different from the URI in `static/userIdDefaultDataType.json`.

File: static/userIdDefaultDataType.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[
+  {
+    "uri": "http://sinopia.io/vocabulary/literalAttribute/userIdDefault",
```
Comment (mjgiarlo) [inline static/userIdDefaultDataType.json:3]: This value was suggested in #3357.

File: static/userIdDefaultDataType.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+    "label": "userid default"
```
Comment (mjgiarlo) [inline static/userIdDefaultDataType.json:4]: Will this be exposed to users? In which case perhaps something more like `user ID default` might be better?

File: static/templates/rt_literal_property_attrs_doc.json [hunk 4]
```diff
@@ -100,6 +100,50 @@
...
+    "@id": "_:b6",
+    "@type": ["http://sinopia.io/vocabulary/PropertyTemplate"],
+    "http://www.w3.org/2000/01/rdf-schema#label": [
+      {
+        "@value": "Literal property attributes"
+      }
+    ],
+    "http://sinopia.io/vocabulary/hasRemark": [
+      {
+        "@value": "Whether literal property is repeatable, required, ordered, etc."
```
Comment (justinlittman) [inline static/templates/rt_literal_property_attrs_doc.json:113]: Needs to be changed.

File: src/components/vocabulary/Vocab.js [hunk 5]
```diff
@@ -36,6 +36,10 @@ const vocabulary = {
     description: "Attributes for a literal",
     url: "http://sinopia.io/vocabulary/hasLiteralAttributes",
   },
+  hasLiteralPropertyAttribute: {
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply src/components/vocabulary/Vocab.js:39]: @justinlittman Hmmm, the example I was basing this off of (from #3357) is `hasPropertyAttribute`: https://github.com/LD4P/sinopia_editor/blob/main/static/templates/rt_property_template_doc.json#L3-L35 But I do see other `has...Attributes` terms below. 

I can go with a plural here, no problem. Does this suggest that `hasPropertyAttribute` should be pluralized at some point too?

File: static/templates/rt_literal_property_attrs_doc.json (hunk 4 above)
Comment (mjgiarlo) [reply static/templates/rt_literal_property_attrs_doc.json:113]: Sure, what text would you like?

File: static/templates/rt_literal_property_attrs_doc.json [hunk 6]
```diff
@@ -100,6 +100,50 @@
...
+    "http://sinopia.io/vocabulary/hasPropertyUri": [
+      {
+        "@id": "http://sinopia.io/vocabulary/hasLiteralPropertyAttribute"
+      }
+    ],
+    "http://sinopia.io/vocabulary/hasPropertyAttribute": [
+      {
+        "@id": "http://sinopia.io/vocabulary/propertyAttribute/repeatable"
+      }
+    ],
```
Comment (mjgiarlo) [inline static/templates/rt_literal_property_attrs_doc.json:125]: @justinlittman I'm also not sure whether this is the right value.

File: src/components/vocabulary/Vocab.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (justinlittman) [reply src/components/vocabulary/Vocab.js:39]: Yes, but no easy way to make that change without breaking everything.

File: static/templates/rt_literal_property_attrs_doc.json (hunk 4 above)
Comment (justinlittman) [reply static/templates/rt_literal_property_attrs_doc.json:113]: "Whether literal property has user ID default."

File: src/components/vocabulary/Vocab.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply src/components/vocabulary/Vocab.js]: @justinlittman Yeah, let's not do that.

File: __tests__/TemplatesBuilder.test.js [hunk 7]
```diff
@@ -142,48 +147,6 @@ _:b2_c14n1 <http://sinopia.io/vocabulary/hasValidationDataType> <http://www.w3.o
     )
   })
 
-  it("builds literal property template without validationDataType value", async () => {
```
Comment (justinlittman) [inline __tests__/TemplatesBuilder.test.js:145]: This test was redundant, so removed.

File: src/TemplatesBuilder.js [hunk 8]
```diff
@@ -147,7 +148,24 @@ export default class TemplatesBuilder {
...
+      propertyTerm,
+      "http://sinopia.io/vocabulary/hasLiteralAttributes"
+    )
+    if (attributeTerm) {
+      propertyTemplate.defaults = this.defaultsForLiteral(attributeTerm)
+      const literalAttrValues = this.valuesFor(
+        this.attributeTerm,
+        "http://sinopia.io/vocabulary/hasLiteralPropertyAttributes"
+      )
+      if (
```
Comment (justinlittman) [inline src/TemplatesBuilder.js:161]: Here's where the user id gets added as a default.

File: src/components/vocabulary/Vocab.js [hunk 9]
```diff
@@ -111,6 +115,11 @@ const vocabulary = {
     description: "Class for a property template",
     url: "http://sinopia.io/vocabulary/PropertyTemplate",
   },
+  "literalPropertyAttribute/userIdDefault": {
```
Comment (ndushay) [inline src/components/vocabulary/Vocab.js:118]: should this start with "has" ?  "hasLiteralAttributeUserIdDefault" ?

**Review by ndushay — CHANGES_REQUESTED** [review]
Looks fine;  couple of quibbles.

File: src/components/vocabulary/Vocab.js (hunk 9 above)
Comment (justinlittman) [reply src/components/vocabulary/Vocab.js:118]: This is referencing the `userIdDefault` in `literalPropertyAttribute` so this is consistent naming.

File: static/templates/rt_literal_property_attrs_doc.json (hunk 6 above)
Comment (justinlittman) [reply static/templates/rt_literal_property_attrs_doc.json:125]: This is a general attribute field; in the future in might contain attributes that are unrelated to defaults.

**Comment by justinlittman** [comment]
@ndushay Thanks for paying attention to the naming. The naming gets really confusing (would make some different choices if starting over...). However, I think the naming in this PR is consistent with other naming in the base templates.

**Review by ndushay — APPROVED** [review]

**Pull request merged**